build_system/src/test.rs

```diff
@@ -494,16 +497,32 @@ fn setup_rustc(env: &mut Env, args: &TestArg) -> Result<(), String> {
     let rust_dir = Some(Path::new("rust"));
     // If the repository was already cloned, command will fail, so doesn't matter.
     let _ = run_command_with_output_and_env(
-        &[&"git", &"clone", &"https://github.com/rust-lang/rust.git"],
+        &[
+            &"git",
+            &"clone",
+            &"https://github.com/rust-lang/rust.git",
+            &"--depth",
+            &"1",
```

Comment:
`--filter=blob:0` is another option which may behave better.

Reply:
I mean in the place of `--depth 1` to make future `git fetch` cheaper. Removing the `git fetch` breaks when the rust repo has already been cloned previously as it may not have the right commit.

Reply:
Sorry, I didn't really understand the usage of this option(nor have I ever used this) and thought it could add up.

Reply:
Thanks for advice.

Reply:
Somehow, the option didn't work as I expected after the demanded replacement.
```
$ git clone https://github.com/rust-lang/rust.git --filter=blob:none

Cloning into 'rust'...
remote: Enumerating objects: 1598065, done.
remote: Counting objects: 100% (1056/1056), done.
remote: Compressing objects: 100% (693/693), done.
Receiving objects:   1% (18171/1598065), 6.85 MiB | 24.00 KiB/s
```
~600MiB
Is it that I misunderstood your advice?

Reply:
Looks like it is way less effective on the rust repo compared to a shallow clone than I expected. (I just tired it and I'm getting 450MB locally) Even `--filter=tree:0` uses 110MB compared to the 30MB for `--depth 1`. Given the huge difference, I did say that using `--depth 1` is fine.

Reply:
The option didn't seem to work as expected. I have probably *misunderstood* your advice.
Please point out my mistake if so. Sorry for wasting your time as a green hand.

Just in case you really mean direct replacement, the following experiments may be of relevance.
With only the `--filter=blob:none`(0 seem to be a wrong choice), the objects remain extremely large:
```
$ git clone https://github.com/rust-lang/rust.git --filter=blob:none
Cloning into 'rust'...
remote: Enumerating objects: 1598065, done.
remote: Counting objects: 100% (1056/1056), done.
remote: Compressing objects: 100% (693/693), done.
Receiving objects:   2% (32526/1598065), 12.48 MiB | 57.00 KiB/s
```
~600MiB

while the larger seem to be smaller.
```
$ git clone https://github.com/rust-lang/rust.git --filter=blob:none --depth=1
Cloning into 'rust'...
remote: Enumerating objects: 3287, done.
remote: Counting objects: 100% (3287/3287), done.
remote: Compressing objects: 100% (2661/2661), done.
remote: Total 3287 (delta 7), reused 2314 (delta 7), pack-reused 0
Receiving objects: 100% (3287/3287), 1.45 MiB | 5.32 MiB/s, done.
Resolving deltas: 100% (7/7), done.
remote: Enumerating objects: 44743, done.
remote: Counting objects: 100% (44743/44743), done.
remote: Compressing objects: 100% (37881/37881), done.
Receiving objects: 100% (44743/44743), 28.86 MiB | 1.65 MiB/s, done.
```
~30MiB

which are both lower than the average size.

As for the real effect of the two options added together, we can see the original download size to be slightly SMALLER than the previous choice:

```
$ git clone https://github.com/rust-lang/rust.git --depth=1
Cloning into 'rust'...
remote: Enumerating objects: 48030, done.
remote: Counting objects: 100% (48030/48030), done.
remote: Compressing objects: 100% (40550/40550), done.
remote: Total 48030 (delta 6420), reused 36610 (delta 5967), pack-reused 0
Receiving objects: 100% (48030/48030), 30.22 MiB | 186.00 KiB/s, done.
Resolving deltas: 100% (6420/6420), done.
```